web/src/page_params.ts

```diff
@@ -1,7 +1,17 @@
 import $ from "jquery";
 
+type ContributorProfile = {
+    avatar: string;
+    email: string;
+    github_username?: string;
+    name: string;
+};
+type RepositoryDetails = {[key: string]: number};
```

Comment:
We may use `Record<string, string>` for better readability, but I think having `RepositoryDetails` here might not be necessary?

Reply:
I have created these types in `teams.ts`, do these look good?
```
const RepoName = ["zulip" , "zulip-desktop" , "zulip-mobile" , "python-zulip-api" , "zulip-js" , "zulipbot" , "zulip-terminal" , "zulip-ios-legacy" , "zulip-android"] as const;
type RepositoryNameType = typeof RepoName[number]

export type Contributor = {
    avatar: string;
    email: string;
    github_username?: string;
    name: string;
} & {
    [K in RepositoryNameType]:number
};
```

Reply:
But this approach makes use of some `as` assertions. I can either leave them like that or use the below type guard:
```
function isRepositoryName(name: string): name is RepositoryNameType {
    return RepositoryName.includes(name as RepositoryNameType)
  }
```

Reply:
Yeah, the types look fine. See the other comments for a workaround to the assertions.